mypy/checkexpr.py

```diff
@@ -789,17 +789,22 @@ def check_typeddict_call_with_kwargs(
         context: Context,
         orig_callee: Type | None,
     ) -> Type:
-        if not (callee.required_keys <= set(kwargs.keys()) <= set(callee.items.keys())):
+        actual_keys = kwargs.keys()
+        found_set = set(actual_keys)
+        if not (callee.required_keys <= found_set <= set(callee.items.keys())):
             expected_keys = [
                 key
                 for key in callee.items.keys()
-                if key in callee.required_keys or key in kwargs.keys()
+                if key in callee.required_keys or key in found_set
             ]
-            actual_keys = kwargs.keys()
             self.msg.unexpected_typeddict_keys(
                 callee, expected_keys=expected_keys, actual_keys=list(actual_keys), context=context
             )
-            return AnyType(TypeOfAny.from_error)
+            if callee.required_keys > found_set:
+                # found_set is not a sub-set of the required_keys
```

Comment:
This comment contradicts the condition, the `if` condition specifically checks that `found_set` _is_ a subset of `required_keys`.

Reply:
Done: https://github.com/python/mypy/commit/f7a7f4f84ac06e5dadc7d6ab05094cd2ca9cd782#diff-42eda644ce5003e33ea0766d7601fcfb4934ad106429abfe115954a1fccddd07R804